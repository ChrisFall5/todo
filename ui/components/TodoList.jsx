import React from "react";
import PropTypes from "prop-types";
import {
  Checkbox,
  List,
  Icon
} from 'semantic-ui-react';

const TodoList = ({ todos, markItemComplete, deleteTodo }) => (
  <List>
    {todos.map((todo) => {
      if (todo.isDeleted) {
        return (
          <List.Item key={todo.id}>
            <Checkbox
              disabled
              className = "todo-text"
              label = {todo.text}
              checked = {todo.isComplete}
            />
          </List.Item>
        );
      }
      return (
        <List.Item key={todo.id}>
          <Checkbox
            className = "todo-text"
            label = {todo.text}
            checked = {todo.isComplete}
            onClick = {() => markItemComplete({ variables: { id: todo.id, complete: !todo.isComplete } })}
          />
          <Icon
            name="delete"
            link
            className="deleteButton"
            onClick={() => deleteTodo ({ variables: { id: todo.id } })}
          />
        </List.Item>
      );
    })}
  </List>
);

TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.object).isRequired,
  markItemComplete: PropTypes.func,
  deleteTodo: PropTypes.func,
};

TodoList.defaultProps = {
  markItemComplete: () => {},
  deleteTodo: () => {},
};

export default TodoList;
