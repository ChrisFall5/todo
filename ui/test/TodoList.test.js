import React from "react";
import Renderer from "react-test-renderer";
import TodoList from "../components/TodoList";
import { nonDeletedTodos, deletedTodos } from "./mocks";

const mockMarkItemComplete = () => {};
const mockDeleteTodo = () => {};

it('renders non-deleted todo list correctly', () => {
  const tree = Renderer.create(<TodoList
    todos={nonDeletedTodos}
    markItemComplete={mockMarkItemComplete}
    deleteTodo={mockDeleteTodo}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders deleted todo list correctly', () => {
  const tree = Renderer.create(<TodoList todos={deletedTodos} />).toJSON();

  expect(tree).toMatchSnapshot();
});
