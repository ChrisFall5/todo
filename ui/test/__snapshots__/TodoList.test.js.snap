// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders deleted todo list correctly 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="ui disabled checkbox todo-text"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        disabled={true}
        readOnly={true}
        tabIndex={-1}
        type="checkbox"
      />
      <label>
        Delete Me
      </label>
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="ui checked disabled checkbox todo-text"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="hidden"
        disabled={true}
        readOnly={true}
        tabIndex={-1}
        type="checkbox"
      />
      <label>
        Test todo input
      </label>
    </div>
  </div>
</div>
`;

exports[`renders non-deleted todo list correctly 1`] = `
<div
  className="ui list"
  role="list"
>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="ui checked checkbox todo-text"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label>
        Test Todo
      </label>
    </div>
    <i
      aria-hidden="true"
      className="delete link icon deleteButton"
      onClick={[Function]}
    />
  </div>
  <div
    className="item"
    onClick={[Function]}
    role="listitem"
  >
    <div
      className="ui checkbox todo-text"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label>
        Test Todo 2
      </label>
    </div>
    <i
      aria-hidden="true"
      className="delete link icon deleteButton"
      onClick={[Function]}
    />
  </div>
</div>
`;
